pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -653,8 +653,9 @@ impl RemoteTimelineClient {
         // to syntactically forbid ? or bail! calls here.
         let no_bail_here = || {
             for name in names {
-                upload_queue.latest_files.remove(name);
-                upload_queue.latest_files_changes_since_metadata_upload_scheduled += 1;
+                if upload_queue.latest_files.remove(name).is_some() {
+                    upload_queue.latest_files_changes_since_metadata_upload_scheduled += 1;
+                }
```

Comment:
Similar thing as to the needs_cleanup, would prefer explicit comment if this is supposed to go the other way around.

After posting this PR I vaguely remember some possible back-and-forth with this.